test/extensions/key_value/file_based/key_value_store_test.cc

```diff
@@ -103,6 +103,33 @@ TEST_F(KeyValueStoreTest, Iterate) {
   EXPECT_EQ(1, stop_early_counter);
 }
 
+#ifndef NDEBUG
+TEST_F(KeyValueStoreTest, ShouldCrashIfIterateCallbackAddsOrUpdatesStore) {
```

Comment:
I see we have a new test for the KeyValueStore. Do we need a new test for the AlternateProtocolsCache, or would the existing AlternateProtocolsCache tests crash with the new key value store implementation?

Reply:
Oh interesting. I think AlternateProtocolsCacheImplTest.CacheLoad arranges for a non-empty store when the cache is created. Is there a way we could EXPECT_CALL() in some fashion which would verify that the insert is not running? (If we do the change to avoid the copies, maybe we can just do: EXPECT_CALL(*store_, addOrUpdate(_, _)).Times(0); before the call to initialize()?)

Reply:
added a test that loads and checks addOrUpdate is called 0 times as suggested.